program/rust/src/tests/test_twap.rs

```diff
@@ -87,12 +96,18 @@ impl PriceCumulative {
     }
     pub fn check_num_down_slots(&self, data: &[DataEvent]) {
         assert_eq!(
-            data.iter()
-                .fold(0, |acc, x| if x.slot_gap > PC_MAX_SEND_LATENCY.into() {
-                    acc + (x.slot_gap - PC_MAX_SEND_LATENCY as u64)
+            data.iter().fold(0, |acc, x| {
+                let latency_threshold = if x.max_latency == 0 {
+                    PC_MAX_SEND_LATENCY.into()
+                } else {
+                    x.max_latency as u64
```

Comment:
`.into()` seems cleaner, also I don't think this pathway is ever explored because DataEvent has max_latency equal to 0 always